packages/adminrouter/extra/src/includes/metrics-location-level.conf

```diff
@@ -0,0 +1,43 @@
+# Output fine-grained location directive level metrics using the VTS module.
+#
+# Caution:
+# A request which matches a location that has a VTS module filter defined will be
+# counted on the Nginx location directive level filter but it will not be counted
+# on a filter defined on the server directive level that it matches as well.
+#
+# The following filter expects a $upstream_tag to be set before importing in order to
+# identify the upstream that the metrics belong to. This constitutes a mechanism to
+# annotate the Admin Router config, allowing one to collect metrics of dynamically
+# resolved upstreams on the fly. It is also the name that will show up in Grafana when
+# referring to a particular upstream. Grouping metrics based on upstream would normally
+# only be possible with Nginx Plus because only then they can be defined in a upstream
+# directive and continue to be dynamically resolved.
+# https://tenzer.dk/nginx-with-dynamic-upstreams/
+#
+# Before including this file, a Nginx variable $upstream_tag must be set like this:
+# set $upstream_tag <CUSTOM_UPSTREAM_TAG>;
+#
+# Create a counter for every backend that requests go to when passing this line in a location directive.
+# Tag the counter with the manually set upstream.
+vhost_traffic_status_filter_by_set_key backend=$upstream_addr ,upstream=${upstream_tag},;
+
+# Create a counter for every status of requests which pass this line in a location directive.
+# Tag the counter with the manually set upstream and the backend that the request was passed on to.
+vhost_traffic_status_filter_by_set_key status=$status ,upstream=${upstream_tag},backend=${upstream_addr},;
+
+# Create a counter for every URI that requests go to when passing this line in a location directive.
+# Tag the counter with the manually set upstream, the backend that the request was passed on to and the resulting status code.
+vhost_traffic_status_filter_by_set_key uri=$uri ,upstream=${upstream_tag},backend=${upstream_addr},status=${status},uri;
+
+# Create a counter for every client that requests where sent from that pass this line in a location directive.
```

Comment:
where -> were, or "for every client that sent requests"

Also, I'm not clear what "pass this line in a location directive" means, here and in previous stanza.

Reply:
Nginx works by matching request properties first to `server` and then `location` directives with the corresponding server name and URI regex.
The term `passing this line` refers to having matched the regex of a particular a directive so that the filter code defined under it is executed. I imagined this as the code flow for handling the request literally passes by this line of the config.
Do you want me to note that in a comment?

Reply:
Maybe just change `pass` to `match`?